jax/_src/lru_cache.py

```diff
@@ -0,0 +1,139 @@
+# Copyright 2024 The JAX Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import annotations
+
+import logging
+import os
+import tensorflow as tf
+from typing import Any
+
+from jax._src import config
+from jax._src.compilation_cache_interface import CacheInterface
+from jax._src.gfile_cache import GFileCacheImpl
+
+
+filelock: Any = None
+try:
+  import filelock
+except ImportError:
+  pass
+
+
+logger = logging.getLogger(__name__)
+
+
+def is_path_gs(path: str) -> bool:
+    return path.startswith("gs://")
+
+
+def get_size(path: str) -> int:
```

Comment:
style nit: "private" functions that are only used in this module should have an underscore, e.g. `_get_size`. Only the functions that are meant to be called by other modules should not have the underscore.

Note this is a different level of visibility than the actual jax public API. This whole module is private from the user API since it's in _src. So the naming convention only matter for internal implementation code. It's less important than keeping the actual public API minimal but still nice to do so it's clear what code can be easily changed without breaking other files.

Reply:
Done because now the functions are removed